modules/mod_ginger_rdf/support/ginger_json_ld.erl

```diff
@@ -239,25 +240,36 @@ compact_predicate(Predicate) ->
 -spec deserialize(tuple() | list()) -> #rdf_resource{}.
 deserialize(JsonLd) when is_map(JsonLd) ->
     Context = maps:get(<<"@context">>, JsonLd, #{}),
+    deserialize_with_context(undefined, JsonLd, Context);
+deserialize(JsonLd) ->
+    %% Fall back to mochijson {struct, ...} structure
+    open(JsonLd).
+
+%% Guide the parser; indicate that we are actually looking for a specific id
+%% so we don't get confused when we encounter multiple subjects in the jsonld.
+deserialize(Id, JsonLd) when is_map(JsonLd) ->
+    Context = maps:get(<<"@context">>, JsonLd, #{}),
+    deserialize_with_context(Id, JsonLd, Context).
+
+deserialize_with_context(Id, JsonLd, Context) ->
+    LocalContext = maps:get(<<"@context">>, JsonLd, #{}),
+    TotalContext = maps:merge(Context, LocalContext),
     maps:fold(
         fun(Key, Value, Acc) ->
-            deserialize(Key, Value, Acc, Context)
+            deserialize(Key, Value, Acc, TotalContext)
         end,
-        #rdf_resource{},
+        #rdf_resource{id = Id},
```

Comment:
This seems an unnecessary addition, I don't see this `Id` being used 🤔 
```suggestion
        #rdf_resource{},
```

Reply:
This is actually the core of the fix ;) - in deserialize the id of the containing rdf_resource is not set, (because) it can be ambiguous in the case of multiple subjects (with id) described, so what we do if we know the uri of the requested resource, is set it explicitly. It might not be used in this local piece of code, but it is essential as identifier of the (top level) rdf_resource.